src/main/scala/com/couchbase/spark/sql/N1QLRelation.scala

```diff
@@ -72,8 +72,23 @@ class N1QLRelation(bucket: String, userSchema: Option[StructType], parameters: M
       .map(_.value.toString)
     val dataset = sqlContext.sparkSession.createDataset(rdd)(Encoders.STRING)
 
+    // datetime and timestamp defaults were taken from sparks Json Parser:
+    // https://github.com/apache/spark/blob/branch-2.3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala#L84
+    val timestampFormat = if (parameters.get("timestampFormat").isDefined) {
+      parameters("timestampFormat")
+    } else {
+      "yyyy-MM-dd'T'HH:mm:ss.SSSXXX"
```

Comment:
if we want to avoid  hardcoding the timestampFormat and dateFormat, would it make sense  to  only set them if present in the parameters, and if not present just don't set the options at all? It would make the code simpler, and we just keep using the spark defaults if not overridden by the user

Reply:
@daschl yup, thats much better.  My scala is rough so I need to figure out how best to do that :-)

Reply:
@daschl actually, would this be too heavy handed - passing through all parameters?

`    val schema = sqlContext
      .read
      .options(parameters)
      .json(dataset)
      .schema`

Reply:
yeah I'm not sure we want to do that, since we don't know if parameters overlap and then we could  run into unintended consequences

The other option would be to pass through parameters, but "remove" the ones we use